languages/ruby/outline.scm

```diff
@@ -1,67 +1,182 @@
+)
 
+; Singleton method definition without modifier, e.g. `def self.foo`
+(body_statement
+    (singleton_method
+        "def" @context
+        object: (_) @context
+        "." @context
+        name: (_) @name) @item
+)
+
+; Singleton method definition with modifier, e.g. `private_class_method def self.foo`
+(body_statement
+    (call
+        method: (identifier) @context
+        arguments: (argument_list
+            (singleton_method
+                "def" @context
+                object: (_) @context
+                "." @context
+                name: (_) @name) @item
+            )) @item
+)
+
+; Module definition, e.g. `module Foo`
 (module
     "module" @context
     name: (_) @name) @item
 
+; Constant assignment
 (assignment left: (constant) @name) @item
 
-; Support Minitest/RSpec symbols
-;
-; Note that `(_)+` is used to capture one more child nodes, meaning it will also include any modifier symbols, like
-; :focus, so that we can easily jump to focused tests
-(call
-    method: (identifier) @run (#any-of? @run "describe" "context" "test" "it")
-    arguments: (argument_list . (_)+) @name
-) @item
+; Class macros such as `alias_method`, `include`, `belongs_to`, `has_many`, `attr_reader`
+(class
+    (body_statement
+        (call
+            method: (identifier) @name
+            arguments: (argument_list . [
+                    (string) @name
+                    (simple_symbol) @name
+                    (scope_resolution) @name
+                    (constant) @name
+                    "," @context
+                ]* [
+                    (string) @name
+                    (simple_symbol) @name
+                    (scope_resolution) @name
+                    (constant) @name
+                ]
+            )
+        ) @item
+    )
+)
+
+; Module macros such as `alias_method`, `include`
+(module
+    (body_statement
+        (call
+            method: (identifier) @name
+            arguments: (argument_list . [
+                    (string) @name
+                    (simple_symbol) @name
+                    (scope_resolution) @name
+                    (constant) @name
+                    "," @context
+                ]* [
+                    (string) @name
+                    (simple_symbol) @name
+                    (scope_resolution) @name
+                    (constant) @name
+                ]
+            )
+        ) @item
+    )
+)
+
+; Class macros without arguments, such as `private`
+(class
+    (body_statement
+        (identifier) @name @item
+    )
+)
+
+(class
+    (body_statement
+        (call
+            method: (identifier) @name
+            !arguments
+        ) @item
+    )
+)
+
+; Module macros without arguments, such as `private`
+(module
+    (body_statement
+        (identifier) @name @item
+    )
+)
+
+(module
+    (body_statement
+        (call
+            method: (identifier) @name
+            !arguments
+        ) @item
+    )
+)
+
+; Root test block
```

Comment:
I don’t think we need to match all macros at the root, but we should specifically match the test blocks.